frontend/src/old-pages/Clusters/Cost.tsx

```diff
@@ -0,0 +1,293 @@
+              const day_data = data_by_instance_type?.map((obj2: any) => {
+                if (obj2.Keys[0] == type) {
+                    total = Number(obj2.Metrics.NetUnblendedCost.Amount) + total
+                    day_info.push({
+                      x: new Date(obj.TimePeriod.Start),
+                      y: Number(parseFloat(obj2.Metrics.NetUnblendedCost.Amount).toFixed(2)),
+                    },)
+                }
+              })
+              if (total > max) {
+                max = total
+              }
+              total = 0
+          })
+          stack_info.push({
+            title: type,
+            type: 'bar',
+            data: day_info,
+          })
+          day_info = []
+    }) 
+    setState(['app', 'cost', cluster_name, 'stack_info'], stack_info);
+    setState(['app', 'cost', cluster_name, 'dates'], dates);
+    setState(['app', 'cost', cluster_name, 'max'], max);
+    }
+    GetGraphData(query_costs, cluster_name, Start, End)
+  }
+
+  const fetchGraphData = (val: any) => {
+    let start_value = ""
+    let end_value = ""
+    if (!val) {
+      let current_date = new Date();
+      let start = new Date(current_date.setDate(current_date.getDate()-7));
+      let start_string = new Date(start).toISOString().split('T');
+      let end_string = new Date().toISOString().split('T');
+      start_value = String(start_string[0]);
+      end_value = String(end_string[0]);
+    } 
+    else {
+      if (val.type === 'relative') {
+        if (val.unit === 'day') {
+          let current_date = new Date()
+          let start = new Date(current_date.setDate(current_date.getDate()-val.amount));
+          let start_string = new Date(start).toISOString().split('T');
+          let end_string = new Date().toISOString().split('T');
+          start_value = String(start_string[0]);
+          end_value = String(end_string[0]);
+        }
+      } 
+      else {
+        const start = new Date(val.startDate).toISOString().split('T');
+        const end = new Date(val.endDate).toISOString().split('T');
+        start_value = String(start[0]);
+        end_value = String(end[0])
+      }
+    }
+    get_graph_data(String(start_value), String(end_value));
+  }
+
+  // These will be called once on loading of page
+  React.useEffect(() => {
+      check_tags()
+      const Default = {
+          key: 'previous-7-days',
+          type: 'relative',
+          unit: 'day',
+          amount: 7,
+        };
+      fetchGraphData(Default);
+      setDateValue(Default);
+  },[])
+
+  return (
+    <Container
+      header={
+        <Header
+          variant="h2"
+          actions={
+            <SpaceBetween
+              direction="horizontal"
+              size="xs"
+            >
+              <DateRangePicker
+              onChange={({ detail }) => {fetchGraphData(detail.value); setDateValue(detail.value); }}
+                value={dateValue}
+                relativeOptions={[
```

Comment:
I'd factor these options out as the actual view here is particularly long and this is just data